mbox/mbox.c

```diff
@@ -913,6 +913,11 @@ static int mbox_mbox_open(struct Mailbox *m)
 
   adata->fp = fopen(mailbox_path(m), "r+");
   if (!adata->fp)
+  {
+    adata->fp = fopen(mailbox_path(m), "r");
+    m->readonly = true;
+  }
+  if (!adata->fp)
   {
     mutt_perror(mailbox_path(m));
     return -1;
```

Comment:
Quick question: This may be handled elsewhere, but do we need to clean up the `m->readonly` in the second `if (!adata->fp)` statement? Perhaps set `m->readonly = false` after opening read-write?

Example situation:
 1. Open mailbox read-write
 2. Open fails
 3. Open mailbox read-only
 4. Open fails; read-only set to true
 5. Reopen mailbox read-write
 6. Open is successful, but `m->readonly = false` from previous attempt.

Reply:
Yes, that sounds reasonable.  Changed.
Ultimately, I think that the read-only-ness needs to be a separate layer.
Perhaps, even, in the MailboxView.